### PR TITLE
docs: nested hps [DET-5819]

### PR DESCRIPTION
**Pull request by liamcli**
## Description
We've had quite a few users ask about support for nested hps.  It was released awhile ago but documentation was sparse.  

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/training-apis/experiment-config.txt [hunk 1]
```diff
@@ -473,27 +473,45 @@ example:
...
-``layer1_dropout``, and ``learning_rate``. ``global_batch_size`` is set to a constant value; the
-other hyperparameters can take on a range of possible values. A hyperparameter's range is configured
-by the ``type`` field of the map; it must be one of ``categorical``, ``double``, ``int``, or
-``log``. More details on these types are given below.
+
+This configuration defines the following hyperparameters:
+
+-  ``global_batch_size``: a constant value
+
+-  ``optimizer_config``: a top level nested hyperparameter with two child hyperparameters
```
Comment (gh-determined-ai) [inline docs/training-apis/experiment-config.txt:501]: If remaining as is, should end description with a colon (:).

File: docs/training-apis/experiment-config.txt [hunk 2]
```diff
@@ -473,27 +473,45 @@ example:
...
-       maxval: 1.0
-       base: 10.0
-
-This configuration defines four hyperparameters: ``global_batch_size``, ``optimizer``,
-``layer1_dropout``, and ``learning_rate``. ``global_batch_size`` is set to a constant value; the
-other hyperparameters can take on a range of possible values. A hyperparameter's range is configured
-by the ``type`` field of the map; it must be one of ``categorical``, ``double``, ``int``, or
-``log``. More details on these types are given below.
+
+This configuration defines the following hyperparameters:
```
Comment (gh-determined-ai) [inline docs/training-apis/experiment-config.txt:497]: I'd rather see this as a table with the following columns: 
field name | type | description (from what follows the bullet list)

**Review by gh-determined-ai — APPROVED** [review]
The entire file needs some rework so 'Approved' for now.

**Comment by liamcli** [comment]
Will keep bullet list for now and revisit as part of larger docs rewrite later.
